src/mmw/apps/modeling/geoprocessing.py

```diff
@@ -47,7 +48,27 @@
 }
 
 
-def histogram(polygons):
+def histogramPoll(jobId):
+    host = settings.GEOP['host']
+    port = settings.GEOP['port']
+    path = '/jobs/' + jobId
+    conn = httplib.HTTPConnection("%s:%s" % (host, port))
+
+    # Try for about two minutes ...
```

Comment:
We may also want to incorporate a `DELETE` of the job once this code gives up so that the geoprocessing work doesn't continue.